framework/src/main/java/org/apache/drill/test/framework/TestDriver.java

```diff
@@ -187,7 +189,47 @@ public int runTests() throws Exception {
       stopwatch.reset().start();
 
       LOG.info("> RUNNING TESTS (ITERATION " + i + ")..");
-      Collections.shuffle(tests, new Random());
+      String queryLogFilename = drillOutputDir + "/queryLog.log";
+      File file = new File(queryLogFilename);
+      if ((cmdParam.repeatRun == true) && (file.exists() && (file.length() > 0))) {
+        // run the tests in the requested order
+        try (BufferedReader reader = new BufferedReader(new FileReader(new File(
+                                     queryLogFilename)))) {
+          List<DrillTest> tests_copy = Lists.newArrayList(tests);
```

Comment:
Here can we do the switch at the end after you build the list? No copy needed just reassign?